ProjFS.Mac/PrjFSKextTests/VirtualizationRootsTests.mm

```diff
@@ -218,6 +219,49 @@ - (void)testRegisterProviderForPath_InsertionSucceeded
     }
 }
 
+- (void)testRegisterProviderForPath_TwoMountPointsInsertionSucceeded
+{
+    const char* path1 = "/Users/test/code/Repo";
+    shared_ptr<vnode> vnode1 = vnode::Create(self->testMountPoint, path1, VDIR);
+    
+    const char* path2 = "/Volumes/Code/Repo";
+    shared_ptr<mount> secondMountPoint = mount::Create();
+    shared_ptr<vnode> vnode2 = vnode::Create(secondMountPoint, path2, VDIR);
+
+    VirtualizationRootResult result = VirtualizationRoot_RegisterProviderForPath(&self->dummyClient, self->dummyClientPid, path1);
+    XCTAssertEqual(result.error, 0);
+    XCTAssertTrue(VirtualizationRoot_IsValidRootHandle(result.root));
+
+    PrjFSProviderUserClient dummyClient2;
+    VirtualizationRootResult result2 = VirtualizationRoot_RegisterProviderForPath(&dummyClient2, 1000, path2);
+    XCTAssertEqual(result2.error, 0);
+    XCTAssertTrue(VirtualizationRoot_IsValidRootHandle(result2.root));
+
+    XCTAssertNotEqual(result.root, result2.root);
+
+    if (VirtualizationRoot_IsValidRootHandle(result.root))
```

Comment:
> Out of curiosity, what would happen if we just let the code inside the `if(...)` run?

`s_virtualizationRoots[result.root].providerUserClient` on the line below would use a negative index into a 0-based array, which we *really* don't want to do even if the test is failing.

`XCTAssert` macros do not exit the test when they fail. And yes, this is a bit weird compared to languages where exceptions are safe to use. (Objective-C exceptions are de facto deprecated because they're almost impossible to get right.)